autoload/ale/command.vim

```diff
@@ -35,6 +36,11 @@ function! ale#command#FormatCommand(buffer, command, pipe_file_if_needed) abort
         let l:command = substitute(l:command, '%s', '\=ale#Escape(l:filename)', 'g')
     endif
 
+    if l:command =~# '%d'
+        let l:dir = fnamemodify(bufname(a:buffer), ':p:h')
+        let l:command = substitute(l:command, '%d', '\=ale#Escape(l:dir)', 'g')
+    endif
```

Comment:
Adding this will slow down processing for every command, and it probably won't be used by too many of them. I recommend just adding the directory into the command directly instead.

Reply:
Actually we don't need this at all. -srcdir is documented as being able to work on the file directly, but when I tested that yesterday it was very slow for some reason. That was probably due to me importing a Windows VM at the same time which made /tmp/ access very slow for some reason :-/ By the time I figured that out I was practically falling asleep as I was rather tired for some reason (too much VimScript?) >_<

Anyway, PR updated and should be good to go now. Sorry for making it more complicated than it had to be.